### DISCUSSION TITLE
Extracting Coordinates of a specific object ID

### DISCUSSION THREAD

**Discussion by Sina7474** [NVlabs/sionna#617] [Q&A] [answered]
Hello,

I'm using Sionna ray tracing to deploy RIS in optimized locations. One issue I'm facing is extracting the maximum height of specific buildings or objects. I know how to get object IDs using the paths.objects function in Sionna, but I'm struggling to find a way to determine the maximum height of those objects.

Could you please help me figure out how to extract this information?

Thank you.

**Comment by jhoydis** [comment]
Hi @Sina7474,

It is currently difficult to do this programmatically. I would recommend loading the scene in Blender, where you can get the heights very easily.

**Comment by merlinND** [comment, marked as answer]
Hello @Sina7474,

Does the following snippet do what you are looking for? Please double-check the results since I didn't verify correctness myself.

```py
import mitsuba as mi
import sionna as sn


def get_max_height_per_shape(scene: mi.Scene, axis: int = 2) -> dict[str, float]:
    heights = {}
    for shape in scene.shapes():
        key = shape.id()
        assert key not in heights
        heights[shape.id()] = shape.bbox().extents()[axis]

    return heights

def main():
    scene = sn.rt.load_scene(sn.rt.scene.etoile)
    heights = get_max_height_per_shape(scene.mi_scene)
    print(heights)


if __name__ == "__main__":
    main()
```

This assumes that the buildings are parallel to "the ground" and that the up axis is z.

Edit: this also assumes that one building is one `Shape`. If one building is split into several parts due to its materials, you would have to compute the height while taking into account the multiple parts. But it should be feasible by looking at `shape.id()` and combining the bounding boxes (`bbox.expand(other)`) of all parts.

**Reply by Sina7474** [reply]
Hello @merlinND 

Thank you so much for the provided snippet code. 

The provided code is working, and the Object IDs from paths.objects align with both your code and Blender. I verified the building heights through visualization in Blender for several cases. Thanks a lot!

**Reply by merlinND** [reply]
Great, I'm glad it helped!

**Reply by potatoess11** [reply]
Hello, I would like to ask if sionna does not support changing the size of the shape of the target object in the simulation scene

**Reply by merlinND** [reply]
Hello @potatoess11,

Please open a new Discussion with a detailed question, including what you have tried already, code examples, etc if relevant.
